go/arrow/extensions/uuid_test.go

```diff
@@ -35,7 +35,7 @@ var testUUID = uuid.New()
 func TestUUIDExtensionBuilder(t *testing.T) {
 	mem := memory.NewCheckedAllocator(memory.DefaultAllocator)
 	defer mem.AssertSize(t, 0)
-	builder := types.NewUUIDBuilder(mem)
+	builder := extensions.NewUUIDBuilder(mem)
```

Comment:
Can we expand the tests a bit? All builder methods don't seem tested, and the actual contents of the built array are also not tested.

Reply:
I've added several more tests and expanded the existing ones.